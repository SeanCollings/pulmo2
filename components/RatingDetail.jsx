import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import { RATINGS_FACES, RATINGS_OPTIONS } from './RatingScale';
import { useTheme } from '../hooks/useTheme';

const RatingDetail = ({ rating }) => {
  const theme = useTheme();

  const iconColour = theme.DARK ? theme.TEXT : theme.BACKGROUND;
  const backgroundColor = theme.DARK ? theme.SECONDARY : theme.SECONDARY;
  const size = 36;
  const opacity = theme.DARK ? 0.87 : 1;

  return (
    <View style={styles.container}>
      <View style={styles.categoryBody}>
        <View
          style={{
            ...styles.ratingSymbol,
            backgroundColor,
            height: size,
            width: size,
            borderRadius: size,
          }}
        >
          <MaterialCommunityIcons
            name={RATINGS_FACES[rating]}
            color={iconColour}
            size={size * 0.8}
            style={{ opacity }}
          />
        </View>
        <View style={styles.ratingValueContainer}>
          <Text style={{ ...styles.ratingText, color: theme.TEXT, opacity }}>
            {RATINGS_OPTIONS[rating]}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {},
  categoryBody: {
    flexDirection: 'row',
    paddingLeft: 10,
    paddingVertical: 10,
  },
  ratingSymbol: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  ratingValueContainer: {
    justifyContent: 'center',
    paddingLeft: 15,
    paddingBottom: 4,
  },
  ratingText: {
    fontFamily: 'tit-light',
    fontSize: 18,
  },
});

export default RatingDetail;
