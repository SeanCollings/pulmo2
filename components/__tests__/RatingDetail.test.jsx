import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup } from 'react-native-testing-library';
import RatingDetail from '../RatingDetail';

describe('RatingDetail - unit test', () => {
  let component;

  afterEach(() => {
    cleanup();
  });

  test('should render', async () => {
    await act(async () => {
      component = create(<RatingDetail rating={0} />);
    });
    expect(component).toMatchSnapshot();
  });
});
