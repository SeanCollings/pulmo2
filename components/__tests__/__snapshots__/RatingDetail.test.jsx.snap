// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingDetail - unit test should render 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#002f56",
          "borderRadius": 36,
          "height": 36,
          "justifyContent": "center",
          "width": 36,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 28.8,
            },
            Object {
              "opacity": 1,
            },
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï±Š
      </Text>
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
          "paddingBottom": 4,
          "paddingLeft": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3e4b57",
            "fontFamily": "tit-light",
            "fontSize": 18,
            "opacity": 1,
          }
        }
      >
        terrible
      </Text>
    </View>
  </View>
</View>
`;
